File: output/deepseek_chat/avatar/direct_translation/java/python/codeforces_637_A.py

import sys
from collections import defaultdict

def main():
    n = int(sys.stdin.readline())
    ara = list(map(int, sys.stdin.readline().split()))
    count = defaultdict(int)
    for num in ara:
        count[num] += 1
    max_count = max(count.values())
    seen = defaultdict(int)
    for num in ara:
        seen[num] += 1
        if seen[num] == max_count:
            print(num)
            break

if __name__ == "__main__":
    main()

File: output/deepseek_chat/avatar/direct_translation/python/java/codeforces_637_A.java

import java.util.*;

public class codeforces_637_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(sc.nextLine());
        String[] s = sc.nextLine().split(" ");
        Map<String, Integer> w = new HashMap<>();
        for (String str : s) {
            w.put(str, w.getOrDefault(str, 0) + 1);
        }
        List<String> reversedS = new ArrayList<>(Arrays.asList(s));
        Collections.reverse(reversedS);
        int maxCount = Collections.max(w.values());
        String result = "";
        int maxIndex = -1;
        for (Map.Entry<String, Integer> entry : w.entrySet()) {
            if (entry.getValue() == maxCount) {
                int index = reversedS.indexOf(entry.getKey());
                if (index > maxIndex) {
                    result = entry.getKey();
                    maxIndex = index;
                }
            }
        }
        System.out.println(result);
    }
}

